src/Appwrite/Platform/Workers/Mails.php

```diff
@@ -67,7 +67,8 @@ public function action(Message $message, Registry $register): void
         $bodyTemplate = Template::fromFile($bodyTemplate);
         $bodyTemplate->setParam('{{body}}', $body, escapeHtml: false);
         foreach ($variables as $key => $value) {
-            $bodyTemplate->setParam('{{' . $key . '}}', $value);
+            // TODO: hotfix for redirect param
+            $bodyTemplate->setParam('{{' . $key . '}}', $value, escapeHtml: $key !== 'redirect');
```

Comment:
Why shouldn't we escape redirect?

Reply:
<img width="1064" alt="Bildschirmfoto 2024-01-11 um 20 43 46" src="https://github.com/appwrite/appwrite/assets/1759475/c22b54d8-c873-44af-aa76-4d9b7cbd5f3a">
source => https://www.php.net/manual/en/function.htmlspecialchars.php

Reply:
`$redirect` should only be a URL and when the worker receives the `$bodyTemplate` it should either be a custom template or one of ours. Ours is the [email-inner-base.tpl](https://github.com/appwrite/appwrite/blob/bd920ead40b50b8be1cec32fcc8299ec2e5454a3/app/config/locale/templates/email-inner-base.tpl) with the following from the following replaced:

https://github.com/appwrite/appwrite/blob/b0a8aac0b7b7511523d8127792342bea9d552b85/app/config/locale/translations/en.json#L7-L11

For this example, it'd be:

```html
<p>Hey {{user}}</p>

<br>

<p>Follow this link to verify your email address.</p>

<a href="{{redirect}}" target="_blank">{{redirect}}</a>

<p>If you didn’t ask to verify this address, you can ignore this message.</p>

<br>

<p>Thanks</p>

<p>{{project}} team</p>
```